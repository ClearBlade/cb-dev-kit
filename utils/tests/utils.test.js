const { generateJs, generateHTML } = require('../generateReactTemplate');
const generateUnitTest = require('../generateUnitTest');


describe('utils', () => {
    const name = 'TestComponent';
    it('Should generate react template', () => {

        const template = {
            JavaScript: generateJs(name),
            HTML: generateHTML(name)
        }
        expect(template).toMatchSnapshot();
    });

    it('Should generate unit test', () => {
        const template = generateUnitTest(name);
        expect(template).toMatchSnapshot();
    });
});