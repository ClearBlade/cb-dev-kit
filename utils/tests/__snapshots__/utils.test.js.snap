// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils Should generate react template 1`] = `
Object {
  "HTML": "<div id='TestComponent-react-widget'></div>",
  "JavaScript": "import React from 'react';
import ReactDOM from 'react-dom';

class TestComponent extends React.Component {
  render() {
    return (
      <div />
    );
  }
}

ReactDOM.render(<TestComponent />, document.getElementById('TestComponent-react-widget'));
export default TestComponent;",
}
`;

exports[`utils Should generate unit test 1`] = `
"// import TestComponent from './index';

describe('TestComponent', () => {
  it('Expect to have unit tests specified', () => {
    expect(false).toEqual(true);
  });
})"
`;
